### PR TITLE
Release: Add v1.14 release notes

### PR DESCRIPTION
**Pull request by iyerr3**

### CODE REVIEW

File: RELEASE_NOTES [hunk 1]
```diff
@@ -9,6 +9,56 @@ commit history located at https://github.com/apache/madlib/commits/master.
...
+    - Fixed step size initialization in MLP based on learning rate policy
+        (MADLIB-1212)
+    - Fixed PCA issue that leads to failure when grouping column is a TEXT type
+        (MADLIB-1215)
+    - Fixed cat levels output in DT when grouping is enabled (MADLIB-1218)
+    - Fixed and simplified initialization of model coefficients in MLP
+    - Removed source table dependency for predicting regression models in MLP
+        (MADLIB-1223)
+    - Print loss of first iteration in MLP (MADLIB-1228)
+
```
Comment (jingyimei) [inline RELEASE_NOTES]: We should mention MADLIB-1209 Neural net related bug fix.

File: RELEASE_NOTES [hunk 2]
```diff
@@ -9,6 +9,56 @@ commit history located at https://github.com/apache/madlib/commits/master.
...
+    - Added newline between file concatenation during PGXN install (MADLIB-1194)
+    - Fixed upgrade issues in knn (MADLIB-1197)
+    - Added fix to ensure RF variable importance are always non-negative
+    - Fixed inconsistency in LDA output and improved usability
+        (MADLIB-1160, MADLIB-1201)
+    - Fixed MLP and RF predict for models trained in earlier versions to
+        ensure misisng optional parameters are given appropriate default values
+        (MADLIB-1207)
+    - Fixed a scenario in DT where no features exist due categorical columns
+        with single level being dropped led to the database crashing
```
Comment (jingyimei) [inline RELEASE_NOTES]: We should mention the other DT/RF bug fixes from MADLIB-1217 & 1219 & 1225.

File: RELEASE_NOTES [hunk 3]
```diff
@@ -9,6 +9,56 @@ commit history located at https://github.com/apache/madlib/commits/master.
...
+MADlib v1.14:
+
+Release Date: 2018-April-28
+
+New features:
+* New module - Balanced datasets: A sampling module to balance classification
+    datasets by resampling using various techniques including undersampling,
+    oversampling, uniform sampling or user-defined proportion sampling
+    (MADLIB-1168)
+* Mini-batch: Added a mini-batch optimizer for MLP and a preprocessor function
```
Comment (jingyimei) [inline RELEASE_NOTES]: Other JIRAs related to this: MADLIB-1220, MADLIB-1224, MADLIB-1226, MADLIB-1227

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/459/

**Comment by jingyimei** [comment]
LGTM

**Comment by fmcquillan99** [comment]
there are 33 JIRAs in 1.14
https://issues.apache.org/jira/projects/MADLIB/versions/12342305

but only about 26 or 27 JIRAs in listed in these release notes.

If that is explainable then +1.

Otherwise check if there is anything left out and add it.

Then you can merge, I don't need to review it again

thanks

**Comment by jingyimei** [comment]
@fmcquillan99 w/ @orhankislal  We did go through JIRAs one by one. Those not mentioned are either no-op or duplicates.

**Comment by iyerr3** [comment]
Thanks @jingyimei and @orhankislal. I'll go ahead and merge this.

**Pull request closed without merging**